clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -637,6 +638,8 @@ private void handleResponse(RequestHeader header, Struct body, long now) {
                 log.trace("Ignoring empty metadata response with correlation id {}.", header.correlationId());
                 this.metadata.failedUpdate(now);
             }
+            for (String topic : response.errorTopics(Errors.UNKNOWN_TOPIC_OR_PARTITION))
+                this.metadata.handleUnknownTopic(topic);
```

Comment:
@hachikuji As @ewencp said, the motivation was to avoid unnecessary metadata fetches in the REST proxy when a topic was deleted and no more messages were ever sent to the topic. At the moment, every time a topic is deleted, our logs are filled with entries about unknown topics even though no one is sending any message to the topic. The error is not cached because it makes sense to refresh metadata for the case where a message is being sent to a topic, but metadata for the topic is not available yet.